src/utils/decorators/input-validation/input-validation.js

```diff
@@ -167,6 +171,93 @@ const InputValidation = ComposedComponent => class Component extends ComposedCom
     info: PropTypes.array
   });
 
+  componentDidUpdate(prevProps) {
+    if (super.componentDidUpdate) { super.componentDidUpdate(prevProps); }
+
+    if (!this.state.valid || this.state.warning || this.state.info) {
+      this.positionElements();
+    }
+  }
+
+  /**
+   * Returns the DOM node of the parent input.
+   *
+   * @method getTarget
+   * @return {DOM node}
+   */
+  getTarget = () => {
+    return ReactDOM.findDOMNode(this._input); // eslint-disable-line react/no-find-dom-node
+  }
+
+  /**
+   * Returns the DOM node of the icon.
+   *
+   * @method getIcon
+   * @return {DOM node}
+   */
+  getIcon = () => {
+    return ReactDOM.findDOMNode(this.validationIcon); // eslint-disable-line react/no-find-dom-node
+  }
+
+  /**
+   * Returns the DOM node of the message.
+   *
+   * @method getMessage
+   * @return {DOM node}
+   */
+  getMessage = () => {
+    return ReactDOM.findDOMNode(this.validationMessage); // eslint-disable-line react/no-find-dom-node
+  }
+
+  /**
+   * Calculates various dimensions needed for render
+   *
+   * @method calculatePosition
+   * @return {Object}
+   */
+  calculatePosition = (icon, message, input) => {
```

Comment:
can we not get away with just using the icon and message (the icon should give position etc)? or is the input essential?

if we don't need the input then we can get rid of the additional ref right?